src/linker/Linker/LinkContext.cs

```diff
@@ -984,5 +984,6 @@ public enum CodeOptimizations
 		RemoveSubstitutions = 1 << 21,
 		RemoveLinkAttributes = 1 << 22,
 		RemoveDynamicDependencyAttribute = 1 << 23,
+		ApplyTypeHierarchyAnnotation = 1 << 24,
```

Comment:
I think the new annotation should have the opposite meaning - if it's turned on, we only apply hierarchy annotations when we see the usage. Library mode would disable such optimization. (It would also fit better with it being an "optimization").

Reply:
Maybe just call it `OptimizeTypeHierarchyAnnotations` or something like that - (and add a comment here describing what is the effect).